pgcli/main.py

```diff
@@ -156,6 +156,10 @@ def cli(database, user, host, port, prompt_passwd, never_prompt):
                     _logger.debug("status: %r", status)
                     output.extend(format_output(rows, headers, status))
                 click.echo_via_pager('\n'.join(output))
+            except KeyboardInterrupt:
+                pgexecute.reconnect()
```

Comment:
Good catch!

My first stab at this showed that the connection was automatically cancelled by `psycopg2.extras.wait_select`.

Looking at the implementation of `wait_select` though, I am not 100% sure WHY it closes the connection. Will test and give you an update.